### PR TITLE
rpmsgblk: rpmsg block device support

### PR DESCRIPTION
**Pull request by CV-Bowen**
## Summary
Like rpmsgdev and rpmsgmtd, rpmsgblk allow the local cpu to access the block device in the remote cpu.

## Impact
New feature

## Testing
test with ramdisk and config sim:rpserver  and sim:rpproxy

### CODE REVIEW

File: drivers/misc/rpmsgblk.c [hunk 1]
```diff
@@ -0,0 +1,1106 @@
...
+  ret = rpmsgblk_geometry(inode, &priv->geo);
+  if (ret < 0)
+    {
+      ferr("Get geometry failed, ret=%d\n", ret);
+      return ret;
+    }
+
+  /* Sanity checks */
+
+  DEBUGASSERT(priv != NULL);
```
Comment (xiaoxiang781216) [inline drivers/misc/rpmsgblk.c:287]: move before line 277

File: boards/sim/sim/sim/src/sim_bringup.c [hunk 2]
```diff
@@ -231,6 +236,16 @@ int sim_bringup(void)
     }
 #endif
 
+#if !defined(CONFIG_DISABLE_MOUNTPOINT) && defined(CONFIG_BLK_RPMSG_SERVER)
+  ramdiskstart = (FAR uint8_t *)kmm_malloc(512 * 2048);
```
Comment (xiaoxiang781216) [inline boards/sim/sim/sim/src/sim_bringup.c:240]: remove FAR

Comment (CV-Bowen) [reply boards/sim/sim/sim/src/sim_bringup.c]: Done

File: drivers/misc/rpmsgblk.c (hunk 1 above)
Comment (CV-Bowen) [reply drivers/misc/rpmsgblk.c]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**